File: prisma/refs/java/pow-x-n.java
class Solution {
    public double myPow(double x, int n) {
        long exp = Math.abs((long) n);
        double base = x, out = 1;
        while (exp > 0) {
            if ((exp & 1) == 1) out *= base;
            base *= base;
            exp >>= 1;
        }
        return n < 0 ? 1 / out : out;
    }
}

File: prisma/refs/py/pow-x-n.py
def my_pow(x, n):
    if n == 0:
        return 1.0
    exp = abs(n)
    base, out = x, 1.0
    while exp:
        if exp & 1:
            out *= base
        base *= base
        exp >>= 1
    return 1 / out if n < 0 else out
